pkg/core/chain/chain.go

```diff
@@ -473,6 +469,20 @@ func (c *Chain) GetSyncProgress(_ context.Context, e *node.EmptyRequest) (*node.
 	return &node.SyncProgressResponse{Progress: float32(100.0)}, nil
 }
 
+// CalculateSyncProgress of the node.
+func (c *Chain) CalculateSyncProgress() float64 {
+	if c.highestSeen == 0 {
```

Comment:
since this method is now used within the `Transactor`, shouldn't `c.highestSeen` and `c.tip` be protected by a read-lock?

Reply:
Good catch, it is definitely needed